src/Nethermind/Nethermind.Trie/TrieNode.cs

```diff
@@ -49,21 +47,72 @@ public partial class TrieNode
         /// </summary>
         public bool IsSealed => !IsDirty;
 
-        public bool IsPersisted { get; set; }
-
         public Keccak? Keccak { get; internal set; }
 
         public byte[]? FullRlp { get; internal set; }
 
-        public NodeType NodeType { get; private set; }
+        private long _value;
+
+        private const long NodeTypeMask = 0b11;
+        private const long DirtyMask = 0b100;
+        private const int DirtyShift = 2;
+        private const long PersistedMask = 0b1000;
+        private const int PersistedShift = 3;
+        private const long HasLastSeenMask = 0b1_0000;
+        private const int HasLastSeenShift = 4;
+        private const int LastSeenShiftRight = 8;
+
+        private const long IsBoundaryProofNodeMask = 0b10_0000;
+        private const int IsBoundaryProofNodeShift = 5;
+
+        // 6, 7 bits are free to use
+
+        public NodeType NodeType
+        {
+            get => (NodeType)(_value & NodeTypeMask);
+            private set => _value = (_value & ~NodeTypeMask) | (long)value;
+        }
+
+        public bool IsDirty
+        {
+            get => (_value & DirtyMask) > 0;
+            private set => _value = (_value & ~DirtyMask) | (value ? 1L << DirtyShift : 0L);
+        }
+
+        public bool IsPersisted
+        {
+            get => (_value & PersistedMask) > 0;
+            set => _value = (_value & ~PersistedMask) | (value ? (1L << PersistedShift) : 0L);
+        }
 
-        public bool IsDirty { get; private set; }
+        public bool IsBoundaryProofNode
+        {
+            get => (_value & IsBoundaryProofNodeMask) > 0;
+            set => _value = (_value & ~IsBoundaryProofNodeMask) | (value ? (1L << IsBoundaryProofNodeShift) : 0L);
+        }
 
         public bool IsLeaf => NodeType == NodeType.Leaf;
         public bool IsBranch => NodeType == NodeType.Branch;
         public bool IsExtension => NodeType == NodeType.Extension;
 
-        public long? LastSeen { get; set; }
+        public long? LastSeen
+        {
+            get
+            {
+                return (_value & HasLastSeenMask) > 0 ? _value >> LastSeenShiftRight : default(long?);
```

Comment:
LastSeen has to be > 0, so if it is 0 it can be treated as null, which removes the need for `HasLastSeenMask`

Reply:
`LastSeen` has been nullable before. I'm aware that they are asserts in regards to `LastSeen has to be > 0`. This looks like a potential trouble maker in the benchmark 

```asm
; not nullable
TrieNode.set_LastSeen(Int64)
    L0000: push edi
    L0001: push esi
    L0002: push ebx
    L0003: lea eax, [ecx+4]
    L0006: mov edx, [eax]
    L0008: mov eax, [eax+4]
    L000b: mov esi, [esp+0x14]
    L000f: mov edi, [esp+0x10]
    L0013: shl edi, 8
    L0016: mov ebx, [esp+0x10]
    L001a: shld esi, ebx, 8
    L001e: or edx, edi
    L0020: or eax, esi
    L0022: add ecx, 4
    L0025: mov [ecx], edx
    L0027: mov [ecx+4], eax
    L002a: pop ebx
    L002b: pop esi
    L002c: pop edi
    L002d: ret 8

; nullable
TrieNode.set_LastSeen(System.Nullable`1<Int64>)
    L0000: push ebp
    L0001: mov ebp, esp
    L0003: push edi
    L0004: push esi
    L0005: push ebx
    L0006: cmp byte ptr [ebp+8], 0
    L000a: jne short L0029
    L000c: lea eax, [ecx+4]
    L000f: mov edx, [eax]
    L0011: mov eax, [eax+4]
    L0014: and edx, 0xffffffef
    L0017: and eax, 0xffffffff
    L001a: add ecx, 4
    L001d: mov [ecx], edx
    L001f: mov [ecx+4], eax
    L0022: pop ebx
    L0023: pop esi
    L0024: pop edi
    L0025: pop ebp
    L0026: ret 0x10
    L0029: lea eax, [ecx+4]
    L002c: mov edx, [eax]
    L002e: mov eax, [eax+4]
    L0031: or edx, 0x10
    L0034: or eax, 0
    L0037: mov esi, [ebp+0x10]
    L003a: mov edi, [ebp+0x14]
    L003d: mov ebx, esi
    L003f: shl ebx, 8
    L0042: shld edi, esi, 8
    L0046: or edx, ebx
    L0048: or eax, edi
    L004a: add ecx, 4
    L004d: mov [ecx], edx
    L004f: mov [ecx+4], eax
    L0052: pop ebx
    L0053: pop esi
    L0054: pop edi
    L0055: pop ebp
    L0056: ret 0x10
```